### PR TITLE
LCD ILI9881C: Supported LCD mirror. Not supported swap XY.

### PR DESCRIPTION
**Pull request by espzav**
# ESP-BSP Pull Request checklist

- [x] Version of modified component bumped
- [x] CI passing

# Change description
Updated LCD ILI9881C initialization, set color mode. Add support for mirror x/y. Not supported swap XY by driver.

### CODE REVIEW

File: components/lcd/esp_lcd_ili9881c/esp_lcd_ili9881c.c [hunk 1]
```diff
@@ -413,34 +409,30 @@ static esp_err_t panel_ili9881c_mirror(esp_lcd_panel_t *panel, bool mirror_x, bo
...
     } else {
-        ili9881c->madctl_val &= ~LCD_CMD_MX_BIT;
+        ili9881c->opmode_val |= (1 << 1);
     }
     if (mirror_y) {
-        ili9881c->madctl_val |= LCD_CMD_MY_BIT;
+        ili9881c->opmode_val |= 1;
     } else {
-        ili9881c->madctl_val &= ~LCD_CMD_MY_BIT;
+        ili9881c->opmode_val &= ~1;
```
Comment (suda-morris) [inline components/lcd/esp_lcd_ili9881c/esp_lcd_ili9881c.c:419]: let's define a macro for this bitfield

File: components/lcd/esp_lcd_ili9881c/esp_lcd_ili9881c.c [hunk 2]
```diff
@@ -276,9 +270,11 @@ esp_err_t esp_lcd_new_panel_ili9881c(const esp_lcd_panel_io_handle_t io, const e
     switch (panel_dev_config->rgb_ele_order) {
     case LCD_RGB_ELEMENT_ORDER_RGB:
         ili9881c->madctl_val = 0;
+        ili9881c->opmode_val |= LCD_CMD_BGR_BIT;
```
Comment (suda-morris) [inline components/lcd/esp_lcd_ili9881c/esp_lcd_ili9881c.c:273]: does the opmode control the RGB element order?

**Comment by espzav** [comment]
@suda-morris Thank you for review. All fixed.

**Review by Lzw655 — COMMENTED** [review]
Hi @espzav, thank you for updating. According to my testing, both the RGB element order and `mirror()` only require the use of the `36h` command, without needing to set `22h`. Could you please confirm this again?

File: components/lcd/esp_lcd_ili9881c/esp_lcd_ili9881c.c [hunk 3]
```diff
@@ -413,34 +409,30 @@ static esp_err_t panel_ili9881c_mirror(esp_lcd_panel_t *panel, bool mirror_x, bo
...
-        ili9881c->madctl_val
+
+    /* Set page 1 */
+    ESP_RETURN_ON_ERROR(esp_lcd_panel_io_tx_param(io, 0xFF, (uint8_t[]) {
+        0x98, 0x81, 0x01
+    }, 3), TAG, "send command failed");
+    ESP_RETURN_ON_ERROR(esp_lcd_panel_io_tx_param(io, 0x22, (uint8_t[]) {
+        ili9881c->opmode_val
     }, 1), TAG, "send command failed");
     return ESP_OK;
```
[the remark this answers is not in the document]
Comment (espzav) [reply components/lcd/esp_lcd_ili9881c/esp_lcd_ili9881c.c:440]: Good point, thank you. I added "set to page 0".

**Comment by espzav** [comment]
> According to my testing, both the RGB element order and `mirror()` only require the use of the `36h` command, without needing to set `22h`. Could you please confirm this again?

Hi @Lzw655, Awesome! It is not in documentation, but it is working (inverted `22h`). I tested before with `LCD_CMD_MX_BIT` and `LCD_CMD_MY_BIT` and it didn't work. With same bits, which was used on `22h` register, it is working. I will change it in driver. Thank you!

**Comment by espzav** [comment]
@tore-espressif PTAL

**Pull request merged**